### ISSUE TITLE
OOM: java.lang.RuntimeException: Could not invoke BlobModule.createFromParts

### ISSUE THREAD

**Issue by wouterds** [joltup/rn-fetch-blob#434]
We're using `rn-fetch-blob` in combination with `redux-persist-filesystem-storage` as storage provider for `redux-persist` in our React Native project. However **on Android** when we're doing frequent store updates (and thus the data is persisted to disk every few seconds) it eventually crashes because **the app runs out of memory**.

**Project info:**
- react: 16.8.6
- react-native: 0.60.5

<details>
<summary><b>crash report</b></summary>
<p>

```text
09-09 15:26:22.186 11746 11763 I art     : Background partial concurrent mark sweep GC freed 26431(1435KB) AllocSpace objects, 23(16MB) LOS objects, 8% free, 174MB/190MB, paused 2.166ms total 130.065ms
09-09 15:26:22.186 11746 11886 I art     : WaitForGcToComplete blocked for 117.065ms for cause Alloc
09-09 15:26:22.186 11746 11886 I art     : Starting a blocking GC Alloc
09-09 15:26:22.186 11746 11852 I art     : WaitForGcToComplete blocked for 81.634ms for cause Alloc
09-09 15:26:22.191 11746 11852 I art     : Starting a blocking GC Alloc
09-09 15:26:23.854 11746 12458 I art     : Waiting for a blocking GC Alloc
09-09 15:26:23.856 11746 12458 I art     : Starting a blocking GC Alloc
09-09 15:26:25.427 11746 11852 I art     : Starting a blocking GC Alloc
09-09 15:26:25.428 11746 11852 I art     : Starting a blocking GC Alloc
09-09 15:26:25.565 11746 11852 I art     : Alloc partial concurrent mark sweep GC freed 11777(4MB) AllocSpace objects, 366(16MB) LOS objects, 8% free, 169MB/185MB, paused 2.044ms total 136.688ms
09-09 15:26:26.117 11746 11763 I art     : Clamp target GC heap from 196MB to 192MB
09-09 15:26:26.117 11746 11763 I art     : Background partial concurrent mark sweep GC freed 19130(914KB) AllocSpace objects, 8(8MB) LOS objects, 5% free, 180MB/192MB, paused 3.448ms total 137.115ms
09-09 15:26:27.338 11746 12458 I art     : Waiting for a blocking GC Alloc
09-09 15:26:27.430 11746 12458 I art     : WaitForGcToComplete blocked for 92.102ms for cause Alloc
09-09 15:26:27.431 11746 12458 I art     : Starting a blocking GC Alloc
09-09 15:26:27.798 11746 12458 I art     : Waiting for a blocking GC Alloc
09-09 15:26:27.934 11746 11763 I art     : Clamp target GC heap from 193MB to 192MB
09-09 15:26:27.934 11746 11763 I art     : Background partial concurrent mark sweep GC freed 6552(3MB) AllocSpace objects, 344(10MB) LOS objects, 7% free, 177MB/192MB, paused 1.831ms total 149.658ms
09-09 15:26:27.934 11746 12458 I art     : WaitForGcToComplete blocked for 136.230ms for cause Alloc
09-09 15:26:27.934 11746 12458 I art     : Starting a blocking GC Alloc
09-09 15:26:29.370 11746 11852 I art     : Starting a blocking GC Alloc
09-09 15:26:29.370 11746 11852 I art     : Starting a blocking GC Alloc
09-09 15:26:29.393 11746 11852 I art     : Alloc sticky concurrent mark sweep GC freed 5456(741KB) AllocSpace objects, 51(2MB) LOS objects, 1% free, 188MB/192MB, paused 1.739ms total 18.371ms
09-09 15:26:29.657 11746 11852 I art     : Waiting for a blocking GC Alloc
09-09 15:26:29.677 11746 11746 I art     : Waiting for a blocking GC Alloc
09-09 15:26:29.713 11746 11763 I art     : Clamp target GC heap from 197MB to 192MB
09-09 15:26:29.713 11746 11763 I art     : Background partial concurrent mark sweep GC freed 9638(442KB) AllocSpace objects, 13(10MB) LOS objects, 5% free, 181MB/192MB, paused 2.197ms total 187.744ms
09-09 15:26:29.713 11746 11852 I art     : WaitForGcToComplete blocked for 56.182ms for cause Alloc
09-09 15:26:29.713 11746 11852 I art     : Starting a blocking GC Alloc
09-09 15:26:29.715 11746 11746 I art     : WaitForGcToComplete blocked for 38.818ms for cause Alloc
09-09 15:26:29.715 11746 11746 I art     : Starting a blocking GC Alloc
09-09 15:26:30.055 11746 11852 I art     : Waiting for a blocking GC Alloc
09-09 15:26:30.079 11746 11852 I art     : WaitForGcToComplete blocked for 24.383ms for cause Alloc
09-09 15:26:30.079 11746 11852 I art     : Starting a blocking GC Alloc
09-09 15:26:31.337 11746 12458 I art     : Waiting for a blocking GC Alloc
09-09 15:26:31.483 11746 11763 I art     : Clamp target GC heap from 197MB to 192MB
09-09 15:26:31.484 11746 11763 I art     : Background partial concurrent mark sweep GC freed 68708(3MB) AllocSpace objects, 44(6MB) LOS objects, 5% free, 181MB/192MB, paused 2.044ms total 165.557ms
09-09 15:26:31.484 11746 12458 I art     : WaitForGcToComplete blocked for 146.667ms for cause Alloc
09-09 15:26:31.484 11746 12458 I art     : Starting a blocking GC Alloc
09-09 15:26:31.786 11746 12458 I art     : Waiting for a blocking GC Alloc
09-09 15:26:31.807 11746 12458 I art     : WaitForGcToComplete blocked for 21.423ms for cause Alloc
09-09 15:26:31.807 11746 12458 I art     : Starting a blocking GC Alloc
09-09 15:26:33.306 11746 11852 I art     : Starting a blocking GC Alloc
09-09 15:26:33.306 11746 11852 I art     : Starting a blocking GC Alloc
09-09 15:26:33.341 11746 11852 I art     : Alloc sticky concurrent mark sweep GC freed 7238(3MB) AllocSpace objects, 274(6MB) LOS objects, 5% free, 182MB/192MB, paused 1.708ms total 33.386ms
09-09 15:26:33.414 11746 11852 I art     : Starting a blocking GC Alloc
09-09 15:26:33.414 11746 11852 I art     : Starting a blocking GC Alloc
09-09 15:26:33.527 11746 11852 I art     : Clamp target GC heap from 197MB to 192MB
09-09 15:26:33.528 11746 11852 I art     : Alloc partial concurrent mark sweep GC freed 2392(453KB) AllocSpace objects, 9(8MB) LOS objects, 5% free, 181MB/192MB, paused 1.861ms total 112.091ms
09-09 15:26:33.749 11746 11852 I art     : Starting a blocking GC Alloc
09-09 15:26:33.749 11746 11852 I art     : Starting a blocking GC Alloc
09-09 15:26:33.770 11746 11852 I art     : Alloc sticky concurrent mark sweep GC freed 11286(509KB) AllocSpace objects, 5(180KB) LOS objects, 2% free, 188MB/192MB, paused 1.281ms total 18.829ms
09-09 15:26:33.770 11746 11852 I art     : Starting a blocking GC Alloc
09-09 15:26:33.876 11746 11852 I art     : Clamp target GC heap from 195MB to 192MB
09-09 15:26:33.876 11746 11852 I art     : Alloc partial concurrent mark sweep GC freed 2801(110KB) AllocSpace objects, 6(8MB) LOS objects, 6% free, 179MB/192MB, paused 1.586ms total 105.529ms
09-09 15:26:35.243 11746 12458 I art     : Waiting for a blocking GC Alloc
09-09 15:26:35.301 11746 12458 I art     : WaitForGcToComplete blocked for 57.952ms for cause Alloc
09-09 15:26:35.301 11746 12458 I art     : Starting a blocking GC Alloc
09-09 15:26:35.492 11746 12458 I art     : Waiting for a blocking GC Alloc
09-09 15:26:35.666 11746 11763 I art     : Clamp target GC heap from 200MB to 192MB
09-09 15:26:35.666 11746 11763 I art     : Background partial concurrent mark sweep GC freed 3778(2MB) AllocSpace objects, 216(4MB) LOS objects, 3% free, 184MB/192MB, paused 2.014ms total 176.727ms
09-09 15:26:35.666 11746 12458 I art     : WaitForGcToComplete blocked for 174.316ms for cause Alloc
09-09 15:26:35.666 11746 12458 I art     : Starting a blocking GC Alloc
09-09 15:26:35.859 11746 12458 I art     : Waiting for a blocking GC Alloc
09-09 15:26:35.893 11746 12458 I art     : WaitForGcToComplete blocked for 33.172ms for cause Alloc
09-09 15:26:35.893 11746 12458 I art     : Starting a blocking GC Alloc
09-09 15:26:37.186 11746 11852 I art     : Starting a blocking GC Alloc
09-09 15:26:37.186 11746 11852 I art     : Starting a blocking GC Alloc
09-09 15:26:37.323 11746 11852 I art     : Clamp target GC heap from 195MB to 192MB
09-09 15:26:37.323 11746 11852 I art     : Alloc partial concurrent mark sweep GC freed 7546(2MB) AllocSpace objects, 168(10MB) LOS objects, 6% free, 179MB/192MB, paused 2.410ms total 137.298ms
09-09 15:26:37.447 11746 11852 I art     : Starting a blocking GC Alloc
09-09 15:26:37.448 11746 11852 I art     : Starting a blocking GC Alloc
09-09 15:26:37.461 11746 11852 I art     : Alloc sticky concurrent mark sweep GC freed 8(184B) AllocSpace objects, 2(1616KB) LOS objects, 2% free, 188MB/192MB, paused 1.800ms total 13.549ms
09-09 15:26:37.641 11746 11882 I art     : Waiting for a blocking GC Alloc
09-09 15:26:37.641 11746 11852 I art     : Waiting for a blocking GC Alloc
09-09 15:26:37.646 11746 11746 I art     : Waiting for a blocking GC Alloc
09-09 15:26:37.766 11746 11763 I art     : Clamp target GC heap from 198MB to 192MB
09-09 15:26:37.767 11746 11763 I art     : Background partial concurrent mark sweep GC freed 11278(469KB) AllocSpace objects, 7(8MB) LOS objects, 4% free, 182MB/192MB, paused 1.922ms total 128.112ms
09-09 15:26:37.767 11746 11882 I art     : WaitForGcToComplete blocked for 126.159ms for cause Alloc
09-09 15:26:37.767 11746 11882 I art     : Starting a blocking GC Alloc
09-09 15:26:37.769 11746 11852 I art     : WaitForGcToComplete blocked for 128.112ms for cause Alloc
09-09 15:26:37.770 11746 11852 I art     : Starting a blocking GC Alloc
09-09 15:26:37.770 11746 11746 I art     : WaitForGcToComplete blocked for 123.901ms for cause Alloc
09-09 15:26:37.770 11746 11746 I art     : Starting a blocking GC Alloc
09-09 15:26:37.925 11746 11852 I art     : Waiting for a blocking GC Alloc
09-09 15:26:37.941 11746 12458 I art     : Waiting for a blocking GC Alloc
09-09 15:26:37.963 11746 11852 I art     : WaitForGcToComplete blocked for 38.208ms for cause Alloc
09-09 15:26:37.963 11746 11852 I art     : Starting a blocking GC Alloc
09-09 15:26:37.964 11746 12458 I art     : WaitForGcToComplete blocked for 23.559ms for cause Alloc
09-09 15:26:37.964 11746 12458 I art     : Starting a blocking GC Alloc
09-09 15:26:38.142 11746 11852 I art     : Waiting for a blocking GC Alloc
09-09 15:26:38.142 11746 11746 I art     : Waiting for a blocking GC Alloc
09-09 15:26:38.242 11746 11763 I art     : Clamp target GC heap from 204MB to 192MB
09-09 15:26:38.242 11746 11763 I art     : Background partial concurrent mark sweep GC freed 19665(977KB) AllocSpace objects, 7(2MB) LOS objects, 2% free, 188MB/192MB, paused 2.075ms total 136.047ms
09-09 15:26:38.242 11746 11852 I art     : WaitForGcToComplete blocked for 100.402ms for cause Alloc
09-09 15:26:38.243 11746 11852 I art     : Starting a blocking GC Alloc
09-09 15:26:38.243 11746 11746 I art     : WaitForGcToComplete blocked for 100.830ms for cause Alloc
09-09 15:26:38.243 11746 11746 I art     : Starting a blocking GC Alloc
09-09 15:26:39.036 11746 12458 I art     : Waiting for a blocking GC Alloc
09-09 15:26:39.074 11746 12458 I art     : WaitForGcToComplete blocked for 38.177ms for cause Alloc
09-09 15:26:39.074 11746 12458 I art     : Starting a blocking GC Alloc
09-09 15:26:39.256 11746 12458 I art     : Waiting for a blocking GC Alloc
09-09 15:26:39.366 11746 11763 I art     : Clamp target GC heap from 204MB to 192MB
09-09 15:26:39.366 11746 11763 I art     : Background partial concurrent mark sweep GC freed 3169(1482KB) AllocSpace objects, 106(2MB) LOS objects, 1% free, 188MB/192MB, paused 1.708ms total 115.264ms
09-09 15:26:39.366 11746 12458 I art     : WaitForGcToComplete blocked for 110.198ms for cause Alloc
09-09 15:26:39.367 11746 12458 I art     : Starting a blocking GC Alloc
09-09 15:26:39.439 11746 12458 I art     : Waiting for a blocking GC Alloc
09-09 15:26:39.478 11746 12458 I art     : WaitForGcToComplete blocked for 38.421ms for cause Alloc
09-09 15:26:39.478 11746 12458 I art     : Starting a blocking GC Alloc
09-09 15:26:39.552 11746 12458 I art     : Waiting for a blocking GC Alloc
09-09 15:26:39.690 11746 11763 I art     : Clamp target GC heap from 204MB to 192MB
09-09 15:26:39.691 11746 11763 I art     : Background partial concurrent mark sweep GC freed 1394(1308KB) AllocSpace objects, 118(2MB) LOS objects, 1% free, 188MB/192MB, paused 1.617ms total 141.326ms
09-09 15:26:39.692 11746 12458 I art     : WaitForGcToComplete blocked for 140.106ms for cause Alloc
09-09 15:26:39.692 11746 12458 I art     : Starting a blocking GC Alloc
09-09 15:26:39.791 11746 12458 I art     : Waiting for a blocking GC Alloc
09-09 15:26:39.795 11746 12458 I art     : Starting a blocking GC Alloc
09-09 15:26:39.886 11746 12458 I art     : Waiting for a blocking GC Alloc
09-09 15:26:40.019 11746 11763 I art     : Clamp target GC heap from 204MB to 192MB
09-09 15:26:40.019 11746 11763 I art     : Background partial concurrent mark sweep GC freed 1348(1425KB) AllocSpace objects, 120(2MB) LOS objects, 2% free, 188MB/192MB, paused 1.983ms total 138.641ms
09-09 15:26:40.020 11746 12458 I art     : WaitForGcToComplete blocked for 134.246ms for cause Alloc
09-09 15:26:40.020 11746 12458 I art     : Starting a blocking GC Alloc
09-09 15:26:41.580 11746 11852 I art     : Waiting for a blocking GC Alloc
09-09 15:26:41.594 11746 11852 I art     : WaitForGcToComplete blocked for 13.824ms for cause Alloc
09-09 15:26:41.594 11746 11852 I art     : Starting a blocking GC Alloc
09-09 15:26:41.644 11746 11852 I art     : Starting a blocking GC Alloc
09-09 15:26:41.644 11746 11852 I art     : Starting a blocking GC Alloc
09-09 15:26:41.760 11746 11852 I art     : Clamp target GC heap from 199MB to 192MB
09-09 15:26:41.762 11746 11852 I art     : Alloc partial concurrent mark sweep GC freed 955(94KB) AllocSpace objects, 9(8MB) LOS objects, 4% free, 183MB/192MB, paused 2.380ms total 116.210ms
09-09 15:26:41.812 11746 11852 I art     : Starting a blocking GC Alloc
09-09 15:26:41.812 11746 11852 I art     : Starting a blocking GC Alloc
09-09 15:26:41.826 11746 11852 I art     : Alloc sticky concurrent mark sweep GC freed 7(160B) AllocSpace objects, 0(0B) LOS objects, 0% free, 191MB/192MB, paused 1.312ms total 13.305ms
09-09 15:26:41.826 11746 11852 I art     : Starting a blocking GC Alloc
09-09 15:26:41.939 11746 11852 I art     : Clamp target GC heap from 207MB to 192MB
09-09 15:26:41.940 11746 11852 I art     : Alloc partial concurrent mark sweep GC freed 85(2184B) AllocSpace objects, 0(0B) LOS objects, 0% free, 191MB/192MB, paused 1.708ms total 112.548ms
09-09 15:26:41.940 11746 11852 I art     : Starting a blocking GC Alloc
09-09 15:26:42.053 11746 11852 I art     : Clamp target GC heap from 207MB to 192MB
09-09 15:26:42.053 11746 11852 I art     : Alloc concurrent mark sweep GC freed 41(1408B) AllocSpace objects, 0(0B) LOS objects, 0% free, 191MB/192MB, paused 1.586ms total 112.335ms
09-09 15:26:42.054 11746 11852 I art     : Forcing collection of SoftReferences for 3MB allocation
09-09 15:26:42.054 11746 11852 I art     : Starting a blocking GC Alloc
09-09 15:26:42.174 11746 11852 I art     : Clamp target GC heap from 207MB to 192MB
09-09 15:26:42.174 11746 11852 I art     : Alloc concurrent mark sweep GC freed 16(424B) AllocSpace objects, 0(0B) LOS objects, 0% free, 191MB/192MB, paused 2.136ms total 119.415ms
09-09 15:26:42.174 11746 11852 W art     : Throwing OutOfMemoryError "Failed to allocate a 3538846 byte allocation with 662768 free bytes and 647KB until OOM"
09-09 15:26:42.174 11746 11852 I art     : Starting a blocking GC Alloc
09-09 15:26:42.174 11746 11852 I art     : Starting a blocking GC Alloc
09-09 15:26:42.188 11746 11852 I art     : Alloc sticky concurrent mark sweep GC freed 9(544B) AllocSpace objects, 0(0B) LOS objects, 0% free, 191MB/192MB, paused 1.434ms total 13.244ms
09-09 15:26:42.188 11746 11852 I art     : Starting a blocking GC Alloc
09-09 15:26:42.304 11746 11852 I art     : Clamp target GC heap from 207MB to 192MB
09-09 15:26:42.304 11746 11852 I art     : Alloc partial concurrent mark sweep GC freed 14(376B) AllocSpace objects, 0(0B) LOS objects, 0% free, 191MB/192MB, paused 1.647ms total 114.959ms
09-09 15:26:42.304 11746 11852 I art     : Starting a blocking GC Alloc
09-09 15:26:42.457 11746 11852 I art     : Clamp target GC heap from 207MB to 192MB
09-09 15:26:42.457 11746 11852 I art     : Alloc concurrent mark sweep GC freed 32(728B) AllocSpace objects, 0(0B) LOS objects, 0% free, 191MB/192MB, paused 1.708ms total 152.160ms
09-09 15:26:42.457 11746 11852 I art     : Forcing collection of SoftReferences for 3MB allocation
09-09 15:26:42.457 11746 11852 I art     : Starting a blocking GC Alloc
09-09 15:26:42.576 11746 11852 I art     : Clamp target GC heap from 207MB to 192MB
09-09 15:26:42.576 11746 11852 I art     : Alloc concurrent mark sweep GC freed 21(624B) AllocSpace objects, 0(0B) LOS objects, 0% free, 191MB/192MB, paused 1.556ms total 118.103ms
09-09 15:26:42.576 11746 11852 I art     : Starting a blocking GC HomogeneousSpaceCompact
09-09 15:26:42.884 11746 11852 I art     : Clamp target GC heap from 207MB to 192MB
09-09 15:26:42.884 11746 11852 I art     : HomogeneousSpaceCompact marksweep + semispace GC freed 6(192B) AllocSpace objects, 0(0B) LOS objects, 0% free, 191MB/192MB, paused 304.107ms total 304.107ms
09-09 15:26:42.884 11746 11852 W art     : Throwing OutOfMemoryError "Failed to allocate a 3538846 byte allocation with 654424 free bytes and 639KB until OOM"
09-09 15:26:42.905 11746 11852 D io.sentry.android.event.helper.AndroidEventBuilderHelper: Proguard UUIDs file not found.
09-09 15:26:43.146 11746 11852 I art     : Waiting for a blocking GC Alloc
09-09 15:26:43.148 11746 11746 I art     : Waiting for a blocking GC Alloc
09-09 15:26:43.206 11746 11763 I art     : Clamp target GC heap from 204MB to 192MB
09-09 15:26:43.206 11746 11852 I art     : WaitForGcToComplete blocked for 60.241ms for cause Alloc
09-09 15:26:43.206 11746 11852 I art     : Starting a blocking GC Alloc
09-09 15:26:43.206 11746 11746 I art     : WaitForGcToComplete blocked for 58.502ms for cause Alloc
09-09 15:26:43.207 11746 11746 I art     : Starting a blocking GC Alloc
09-09 15:26:43.216 11746 11852 W System.err: SLF4J: Failed to load class "org.slf4j.impl.StaticMDCBinder".
09-09 15:26:43.216 11746 11852 W System.err: SLF4J: Defaulting to no-operation MDCAdapter implementation.
09-09 15:26:43.216 11746 11852 W System.err: SLF4J: See http://www.slf4j.org/codes.html#no_static_mdc_binder for further details.
--------- beginning of crash
09-09 15:26:43.562 11746 11852 E AndroidRuntime: FATAL EXCEPTION: mqt_native_modules
09-09 15:26:43.562 11746 11852 E AndroidRuntime: Process: io.getdelta.android, PID: 11746
09-09 15:26:43.562 11746 11852 E AndroidRuntime: java.lang.RuntimeException: Could not invoke BlobModule.createFromParts
09-09 15:26:43.562 11746 11852 E AndroidRuntime: 	at com.facebook.react.bridge.JavaMethodWrapper.invoke(JavaMethodWrapper.java:383)
09-09 15:26:43.562 11746 11852 E AndroidRuntime: 	at com.facebook.react.bridge.JavaModuleWrapper.invoke(JavaModuleWrapper.java:158)
09-09 15:26:43.562 11746 11852 E AndroidRuntime: 	at com.facebook.react.bridge.queue.NativeRunnable.run(Native Method)
09-09 15:26:43.562 11746 11852 E AndroidRuntime: 	at android.os.Handler.handleCallback(Handler.java:751)
09-09 15:26:43.562 11746 11852 E AndroidRuntime: 	at android.os.Handler.dispatchMessage(Handler.java:95)
09-09 15:26:43.562 11746 11852 E AndroidRuntime: 	at com.facebook.react.bridge.queue.MessageQueueThreadHandler.dispatchMessage(MessageQueueThreadHandler.java:29)
09-09 15:26:43.562 11746 11852 E AndroidRuntime: 	at android.os.Looper.loop(Looper.java:154)
09-09 15:26:43.562 11746 11852 E AndroidRuntime: 	at com.facebook.react.bridge.queue.MessageQueueThreadImpl$4.run(MessageQueueThreadImpl.java:232)
09-09 15:26:43.562 11746 11852 E AndroidRuntime: 	at java.lang.Thread.run(Thread.java:761)
09-09 15:26:43.562 11746 11852 E AndroidRuntime: Caused by: java.lang.reflect.InvocationTargetException
09-09 15:26:43.562 11746 11852 E AndroidRuntime: 	at java.lang.reflect.Method.invoke(Native Method)
09-09 15:26:43.562 11746 11852 E AndroidRuntime: 	at com.facebook.react.bridge.JavaMethodWrapper.invoke(JavaMethodWrapper.java:372)
09-09 15:26:43.562 11746 11852 E AndroidRuntime: 	... 8 more
09-09 15:26:43.562 11746 11852 E AndroidRuntime: Caused by: java.lang.OutOfMemoryError: Failed to allocate a 3538846 byte allocation with 654424 free bytes and 639KB until OOM
09-09 15:26:43.562 11746 11852 E AndroidRuntime: 	at java.nio.HeapByteBuffer.<init>(HeapByteBuffer.java:54)
09-09 15:26:43.562 11746 11852 E AndroidRuntime: 	at java.nio.HeapByteBuffer.<init>(HeapByteBuffer.java:49)
09-09 15:26:43.562 11746 11852 E AndroidRuntime: 	at java.nio.ByteBuffer.allocate(ByteBuffer.java:261)
09-09 15:26:43.562 11746 11852 E AndroidRuntime: 	at com.facebook.react.modules.blob.BlobModule.createFromParts(BlobModule.java:336)
09-09 15:26:43.562 11746 11852 E AndroidRuntime: 	... 10 more
09-09 15:26:44.297 11746 11746 D ReactNative: ReactInstanceManager.detachViewFromInstance()
09-09 15:26:44.307 11746 11746 W unknown:ReactNative: Receiver already unregistered
09-09 15:26:44.307 11746 11746 W unknown:ReactNative: java.lang.IllegalArgumentException: Receiver not registered: org.wonday.orientation.OrientationModule$2@7fd603c
09-09 15:26:44.307 11746 11746 W unknown:ReactNative: 	at android.app.LoadedApk.forgetReceiverDispatcher(LoadedApk.java:1007)
09-09 15:26:44.307 11746 11746 W unknown:ReactNative: 	at android.app.ContextImpl.unregisterReceiver(ContextImpl.java:1330)
09-09 15:26:44.307 11746 11746 W unknown:ReactNative: 	at android.content.ContextWrapper.unregisterReceiver(ContextWrapper.java:608)
09-09 15:26:44.307 11746 11746 W unknown:ReactNative: 	at org.wonday.orientation.OrientationModule.onHostDestroy(OrientationModule.java:375)
09-09 15:26:44.307 11746 11746 W unknown:ReactNative: 	at com.facebook.react.bridge.ReactContext.onHostDestroy(ReactContext.java:252)
09-09 15:26:44.307 11746 11746 W unknown:ReactNative: 	at com.facebook.react.ReactInstanceManager.moveToBeforeCreateLifecycleState(ReactInstanceManager.java:688)
09-09 15:26:44.307 11746 11746 W unknown:ReactNative: 	at com.facebook.react.ReactInstanceManager.onHostDestroy(ReactInstanceManager.java:597)
09-09 15:26:44.307 11746 11746 W unknown:ReactNative: 	at com.facebook.react.ReactInstanceManager.onHostDestroy(ReactInstanceManager.java:611)
09-09 15:26:44.307 11746 11746 W unknown:ReactNative: 	at com.facebook.react.ReactActivityDelegate.onDestroy(ReactActivityDelegate.java:122)
09-09 15:26:44.307 11746 11746 W unknown:ReactNative: 	at com.facebook.react.ReactActivity.onDestroy(ReactActivity.java:70)
09-09 15:26:44.307 11746 11746 W unknown:ReactNative: 	at android.app.Activity.performDestroy(Activity.java:6886)
09-09 15:26:44.307 11746 11746 W unknown:ReactNative: 	at android.app.Instrumentation.callActivityOnDestroy(Instrumentation.java:1154)
09-09 15:26:44.307 11746 11746 W unknown:ReactNative: 	at android.app.ActivityThread.performDestroyActivity(ActivityThread.java:4227)
09-09 15:26:44.307 11746 11746 W unknown:ReactNative: 	at android.app.ActivityThread.handleDestroyActivity(ActivityThread.java:4258)
09-09 15:26:44.307 11746 11746 W unknown:ReactNative: 	at android.app.ActivityThread.-wrap6(ActivityThread.java)
09-09 15:26:44.307 11746 11746 W unknown:ReactNative: 	at android.app.ActivityThread$H.handleMessage(ActivityThread.java:1557)
09-09 15:26:44.307 11746 11746 W unknown:ReactNative: 	at android.os.Handler.dispatchMessage(Handler.java:102)
09-09 15:26:44.307 11746 11746 W unknown:ReactNative: 	at android.os.Looper.loop(Looper.java:154)
09-09 15:26:44.307 11746 11746 W unknown:ReactNative: 	at android.app.ActivityThread.main(ActivityThread.java:6186)
09-09 15:26:44.307 11746 11746 W unknown:ReactNative: 	at java.lang.reflect.Method.invoke(Native Method)
09-09 15:26:44.307 11746 11746 W unknown:ReactNative: 	at com.android.internal.os.ZygoteInit$MethodAndArgsCaller.run(ZygoteInit.java:889)
09-09 15:26:44.307 11746 11746 W unknown:ReactNative: 	at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:779)
09-09 15:26:44.448 11746 11801 I CrashlyticsCore: Crashlytics report upload complete: 5D765227017C-0001-2DE2-2579187F403D
```
</p>
</details>

**Comment by wouterds** [comment]
Our redux store sometimes gets corrupted when this happens. But can confirm that using the patch here https://github.com/joltup/rn-fetch-blob/pull/432 prevents it from getting corrupted when the app goes OOM & crashes.

**Comment by wouterds** [comment]
The memory leak was not related to this package, this was just the first one to crash because of it. It was a bug in React Native itself and has been fixed here https://github.com/facebook/react-native/commit/1f66ff1199dd8bb0a29954f73cb15d0269b422b4. Our corrupted files was an issue here but glad to see that has been merged in #432!
